pkg/scheduler/framework/v1alpha1/framework.go

```diff
@@ -198,6 +198,10 @@ func (f *framework) QueueSortFunc() LessFunc {
 // cycle is aborted.
 func (f *framework) RunPreFilterPlugins(
 	state *CycleState, pod *v1.Pod) *Status {
+	if pod == nil {
+		return nil
+	}
```

Comment:
Yes, this is because panic does occur, in unit testing:

you can run

`make test WHAT=./pkg/scheduler/framework/v1alpha1/`

```
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x28 pc=0x178eb08]

goroutine 79 [running]:
testing.tRunner.func1(0xc0001e7000)
	/usr/local/Cellar/go/1.13.1/libexec/src/testing/testing.go:874 +0x3a3
panic(0x1859120, 0x203a5f0)
	/usr/local/Cellar/go/1.13.1/libexec/src/runtime/panic.go:679 +0x1b2
k8s.io/kubernetes/pkg/scheduler/framework/v1alpha1.(*framework).RunPreFilterPlugins(0xc000200dc0, 0x0, 0x0, 0x0)
	/Users/wangguoliang/Documents/mos/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/scheduler/framework/v1alpha1/framework.go:214 +0x138
k8s.io/kubernetes/pkg/scheduler/framework/v1alpha1.TestPreFilterPlugins.func3(0xc0001e7000)
	/Users/wangguoliang/Documents/mos/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/scheduler/framework/v1alpha1/framework_test.go:493 +0x10b
testing.tRunner(0xc0001e7000, 0xc0001e3080)
	/usr/local/Cellar/go/1.13.1/libexec/src/testing/testing.go:909 +0xc9
created by testing.(*T).Run
	/usr/local/Cellar/go/1.13.1/libexec/src/testing/testing.go:960 +0x350
FAIL	k8s.io/kubernetes/pkg/scheduler/framework/v1alpha1	0.021s
```

Reply:
FYI:
https://github.com/kubernetes/kubernetes/blob/22ed5c2fe2a4f5da37ecedb58709a4b778b4658e/pkg/scheduler/framework/v1alpha1/framework_test.go#L493

Reply:
Why does the test call RunPrefilterPlugins with nil pod?

Reply:
FYI:https://github.com/kubernetes/kubernetes/commit/37b9e6d1eadb

Reply:
No.

Reply:
@wgliang , can we update the tests and get rid of the nil checks here? Thanks.

Reply:
done